src/coreComponents/constitutive/solid/TriaxialDriver.hpp

```diff
@@ -0,0 +1,205 @@
+                        integer const GEOSX_UNUSED_PARAM( eventCounter ),
+                        real64 const GEOSX_UNUSED_PARAM( eventProgress ),
+                        DomainPartition & GEOSX_UNUSED_PARAM( domain ) ) override;
+
+  /**
+   * @brief Run a strain-controlled test using loading protocol in table
+   * @param solid Solid constitutive model
+   * @param table Table with stress / strain time history
+   */
+  template< typename SOLID_TYPE >
+  void runStrainControlTest( SOLID_TYPE & solid, arrayView2d< real64 > & table );
+
+  /**
+   * @brief Run a mixed stress/strain-controlled test using loading protocol in table
+   * @param solid Solid constitutive model
+   * @param table Table with stress / strain time history
+   */
+  template< typename SOLID_TYPE >
+  void runMixedControlTest( SOLID_TYPE & solid, arrayView2d< real64 > & table );
+
+  /**
+   * @brief Ouput table to file for easy plotting
+   */
+  void outputResults();
+
+  /**
+   * @brief Read in a baseline table from file and compare with computed one (for unit testing purposes)
+   */
+  void compareWithBaseline();
+
+private:
+
+  /**
+   * @struct viewKeyStruct holds char strings and viewKeys for fast lookup
+   */
+  struct viewKeyStruct
+  {
+    constexpr static char const * solidMaterialNameString() { return "material"; }
+    constexpr static char const * modeString() { return "mode"; }
+    constexpr static char const * strainFunctionString() { return "strainFunction"; }
+    constexpr static char const * stressFunctionString() { return "stressFunction"; }
+    constexpr static char const * numStepsString() { return "steps"; }
+    constexpr static char const * outputString() { return "output"; }
+    constexpr static char const * baselineString() { return "baseline"; }
+  };
+
+  int m_numSteps;              ///< Number of load steps
+  string m_solidMaterialName;  ///< Material identifier
+  string m_mode;               ///< Test mode: triaxial, volumetric, oedometer
+  string m_strainFunctionName; ///< Time-dependent function controlling strain (role depends on test mode)
+  string m_stressFunctionName; ///< Time-dependent function controlling stress (role depends on test mode)
+  string m_outputFile;         ///< Output file (optional, no output if not specified)
+  Path m_baselineFile;         ///< Baseline file (optional, for unit testing of solid models)
+  array2d< real64 > m_table;   ///< Table storing time-history of axial/radial stresses and strains
+
+  static localIndex const m_numColumns = 9; ///< Number of columns in data table
+  enum columnKeys { TIME, EPS0, EPS1, EPS2, SIG0, SIG1, SIG2, ITER, NORM }; ///< Enumeration of column keys
+
+  static constexpr localIndex m_maxIter = 25;
+  static constexpr real64 m_newtonTol = 1e-6;
+  static constexpr real64 m_baselineTol = 1e-3;
+};
+
+
+template< typename SOLID_TYPE >
+void TriaxialDriver::runStrainControlTest( SOLID_TYPE & solid, arrayView2d< real64 > & table )
```

Comment:
if this is only called from the source file, it can be defined in the source file instead of the header.